src/main/java/com/onarandombox/MultiverseCore/commands/ListCommand.java

```diff
@@ -0,0 +1,126 @@
+package com.onarandombox.MultiverseCore.commands;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import co.aikar.commands.BukkitCommandIssuer;
+import co.aikar.commands.InvalidCommandArgument;
+import co.aikar.commands.annotation.CommandAlias;
+import co.aikar.commands.annotation.CommandCompletion;
+import co.aikar.commands.annotation.CommandPermission;
+import co.aikar.commands.annotation.Description;
+import co.aikar.commands.annotation.Subcommand;
+import co.aikar.commands.annotation.Syntax;
+import com.onarandombox.MultiverseCore.MultiverseCore;
+import com.onarandombox.MultiverseCore.api.MVWorld;
+import com.onarandombox.MultiverseCore.commandtools.flags.CommandFlagGroup;
+import com.onarandombox.MultiverseCore.commandtools.flags.CommandValueFlag;
+import com.onarandombox.MultiverseCore.commandtools.flags.ParsedCommandFlags;
+import com.onarandombox.MultiverseCore.display.ContentDisplay;
+import com.onarandombox.MultiverseCore.display.filters.ContentFilter;
+import com.onarandombox.MultiverseCore.display.filters.DefaultContentFilter;
+import com.onarandombox.MultiverseCore.display.filters.RegexContentFilter;
+import com.onarandombox.MultiverseCore.display.handlers.PagedSendHandler;
+import com.onarandombox.MultiverseCore.display.parsers.ListContentProvider;
+import org.bukkit.ChatColor;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+@CommandAlias("mv")
+public class ListCommand extends MultiverseCoreCommand {
+    public ListCommand(@NotNull MultiverseCore plugin) {
+        super(plugin);
+
+        registerFlagGroup(CommandFlagGroup.builder("mvlist")
+                .add(CommandValueFlag.builder("--filter", ContentFilter.class)
+                        .addAlias("-f")
+                        .context((value) -> {
+                            try {
+                                return RegexContentFilter.fromString(value);
+                            } catch (IllegalArgumentException e) {
+                                throw new InvalidCommandArgument("Invalid filter: " + value);
+                            }
+                        })
+                        .build())
+                .add(CommandValueFlag.builder("--page", Integer.class)
+                        .addAlias("-p")
+                        .context((value) -> {
+                            try {
+                                return Integer.parseInt(value);
+                            } catch (NumberFormatException e) {
+                                throw new InvalidCommandArgument("Invalid page number: " + value);
+                            }
+                        })
+                        .build())
+                .build());
+    }
+
+    @Subcommand("list")
+    @CommandPermission("multiverse.core.list.worlds")
+    @CommandCompletion("@flags:groupName=mvlist")
+    @Syntax("--filter [filter] --page [page]")
```

Comment:
In my opinion, it might be better to have page as the first argument, not a flag. then allow the --filter argument so it becomes this
```java
@CommandCompletion("@range:1-10|@flags:groupName=mvlist @flags:groupName=mvlist")
```

Reply:
@zax71 can you explain what that does? I'm not super familiar with the syntax.